components/messages.js

```diff
@@ -1,59 +1,42 @@
 import React from "react";
 import route from "can-route";
-import DefineMap from "can-define/map/";
-import Component from "react-view-model/component";
-import PromiseViewModel from "react-view-model/helpers/promise";
 import Message from "../models/message";
+import { connect, ObserveObject } from "ylem";
+import { getAsync } from 'ylem/property-decorators';
 
-export const ViewModel = DefineMap.extend('MessagesVM', {
-	messagesPromise: {
-		Type: PromiseViewModel,
-		value: () => Message.getList({}),
-	},
-	name: {
-		type: "string",
-		value: ""
-	},
-	body: {
-		type: "string",
-		value: ""
+
+
+class Store extends ObserveObject {
+	@getAsync
+	get messages() {
+		return Message.getList({});
+	}
+	messageInfo = {
+		name: "",
+		body: ""
 	}
-});
 
-export default class Messages extends Component {
-	send(e) {
+	send = e => {
 		e && e.preventDefault();
-
 		new Message({
-			name: this.viewModel.name,
-			body: this.viewModel.body
+			name: this.messageInfo.name,
+			body: this.messageInfo.body
```

Comment:
since you've put `messageInfo` as a child object, why not just pass `new Message(this.messageInfo)`?